dashboard/app/controllers/sessions_controller.rb

```diff
@@ -8,6 +8,16 @@ def new
     super
   end
 
+  # POST /resource/sign_in
+  def create
+    super
+    return unless signed_in?
+    return unless current_user.current_sign_in_ip
+    return if UserGeo.find_by_user_id(current_user.id)
+
+    UserGeo.new.populate(current_user.id, current_user.current_sign_in_ip)
```

Comment:
I think this might break the response returned by the `SessionsController#create` action unless you pass the overridden logic as a block to `super` (see [related SO answer](http://stackoverflow.com/a/34816998/2518355)), like this:

```
  # POST /login
  def create
    super do |user|
      # custom code here
      return unless user.current_sign_in_ip # ...etc
    end
  end
```